include/jwt-cpp/jwt.h

```diff
@@ -396,6 +398,61 @@ namespace jwt {
 			return res;
 		}
 
+		/**
+		 * \brief Convert the certificate provided as base64 DER to PEM.
+		 *
+		 * This is useful when using with JWKs as x5c claim is encoded as base64 DER. More info
+		 * (here)[https://tools.ietf.org/html/rfc7517#section-4.7]
+		 *
+		 * \param cert_base64_der_str String containing the certificate encoded as base64 DER
+		 * \param ec		error_code for error_detection (gets cleared if no error occures)
+		 */
+		inline
+		std::string convert_base64_der_to_pem(const std::string& cert_base64_der_str, std::error_code& ec) {
+			ec.clear();
+			const auto decodedStr = base::decode<alphabet::base64>(base::pad<alphabet::base64>(cert_base64_der_str));
+			auto c_str = reinterpret_cast<const unsigned char *> (decodedStr.c_str());
+
+			std::unique_ptr<X509, decltype(&X509_free)> cert(d2i_X509(NULL, & c_str, decodedStr.size()), X509_free);
+			std::unique_ptr<BIO, decltype(&BIO_free_all)> certbio(BIO_new(BIO_s_mem()), BIO_free_all);
+			if(!cert || !certbio) {
+				ec = error::rsa_error::create_mem_bio_failed;
+				return {};
+			}
+
+			if(!PEM_write_bio_X509(certbio.get(), cert.get()))
+			{
+				ec = error::rsa_error::write_cert_failed;
+				return {};
+			}
+
+			char* ptr = nullptr;
+			const uint64_t len = BIO_get_mem_data(certbio.get(), &ptr);
+			if(len <= 0 || ptr == nullptr) {
```

Comment:
This check is wrong as len is unsigned and thus can never be less than 0.
Actually I am a bit surprised the compiler does not warn about this.

Reply:
ah, good catch, so I suppose I can't use an unsigned as `BIO_get_mem_data` could return a negative value in case of an error. I'll cast len below in the string constructor.